### PR TITLE
feat: file transfers analytics rules based on threshold

### PR DESCRIPTION
**Pull request by juju4**
Change(s):
   - Add file transfer analytics rules

   Reason for Change(s):
   - More data exfiltration detection based on static threshold to adapt to context
   - Adapted for OpenSSH SFTP, Sharepoint and Progress MoveIt logs

   Version Updated:
   - No (new rules, starting 1.0.0)

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Pending CI

Note: for moveit, I put in Windows Forwarded Event folder but not sure if best. In my case, logs are in Events table collected through AMA.

### CODE REVIEW

**Comment by v-rbajaj** [comment]
Hi @juju4, please shorten the description of all the analytical rules to 255 characters only.

**Comment by juju4** [comment]
description was inspired of Syslog/Analytic Rules/ssh_potentialBruteForce.yaml and many other yaml files seem to be above 255.

If too long, where do we put other information? any extended description or note field?
mostly the entity mapping for array limitation information.

Also how to add grouping (aggregationKind) on selected entities?
If not already, it would be nice if the yaml rule schema was documented somewhere public as it does not seem to be 1:1 mapping of the ARM template.

**Comment by v-rbajaj** [comment]
Hi @juju4, we acknowledge your concern, let me get back to you by 10 Oct 2023.

**Comment by v-rbajaj** [comment]
Hi @juju4, sorry for delay in response. 
To answer your first question, the limit for 255 characters is only for Hunting queries and not for Analytical rules, so please ignore that. 

Regarding second question, we will get back to you by 12 Oct 2023.

**Comment by v-rbajaj** [comment]
> Also how to add grouping (aggregationKind) on selected entities? If not already, it would be nice if the yaml rule schema was documented somewhere public as it does not seem to be 1:1 mapping of the ARM template.

Hi @juju4, for this question, you can refer to this document - https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide
image

Apart from that, I could there are some validations which are failing in the PR. 
To fix them, please change the extensions of files Solutions/Windows Forwarded Events/Analytic Rules/moveit_file_transfer_above_threshold.yml, Solutions/Windows Forwarded Events/Analytic Rules/moveit_file_transfer_folders_above_threshold.yml to .yaml

**Comment by juju4** [comment]
file extension renaming done.

On [eventGroupingSettings](https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#eventgroupingsettings), this is not the grouping that I was looking for.
I was searching for setting to group repeated alerts based on selected entities, typically Hostname and Account.
Not grouping based on multiple KQL results.
Would this be exist in the yaml definition?

In ARM template, it would be groupingConfiguration as per https://learn.microsoft.com/en-us/azure/templates/microsoft.securityinsights/2022-01-01-preview/alertrules?pivots=deployment-language-bicep#groupingconfiguration with matchingMethod = Selected

File: Solutions/Microsoft 365/Analytic Rules/sharepoint_file_transfer_folders_above_threshold.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+queryFrequency: 15m
+queryPeriod: 15m
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Exfiltration
+relevantTechniques:
+  - T1020
+query: |
+  let threshold = 500;
```
Comment (v-rbajaj) [inline Solutions/Microsoft 365/Analytic Rules/sharepoint_file_transfer_folders_above_threshold.yaml:22]: Hi, can you please fix the query here?
The error we are encountering is in KQL validation due to the fact that Kusto  doesn't allow us to use custom names in the summarize clause without wrapping them in aggregation functions like count() or make_set().
```
let threshold = 500;
OfficeActivity
| where EventSource == "SharePoint" and OfficeWorkload has_any("SharePoint", "OneDrive") and Operation has_any ("FileDownloaded", "FileSyncDownloadedFull", "FileSyncUploadedFull", "FileUploaded")
| summarize count_distinct_SourceRelativeUrl = count_distinct(SourceRelativeUrl), dirlist = make_set(SourceRelativeUrl) by UserId, ClientIP, UserAgent, bin(TimeGenerated, 15m)
| where count_distinct_SourceRelativeUrl >= threshold
| extend DirSample = iff(array_length(dirlist) == 1, tostring(dirlist[0]), strcat("SeeDirListField", "_", tostring(hash(tostring(dirlist))))

```

Comment (juju4) [reply Solutions/Microsoft 365/Analytic Rules/sharepoint_file_transfer_folders_above_threshold.yaml:22]: I believe that if not setting name I was getting another error.
Only seeing this type of error
```
  Error Message:
   Template Id: 8b4f03e7-3460-4401-824d-e65a8dd464f0 is not valid in Line: 4 col: 43
                    Errors: The name 'count_distinct' does not refer to any known function., Code: 'KS211', Severity: 'Error', Location: '275..289'
```

count_distinct is valid - https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/count-distinct-aggfunction
if using count(), altering logic
without name, how do I check threshold?
and using in my prod deployment (deployed from ARM template)

Comment (v-rbajaj) [reply Solutions/Microsoft 365/Analytic Rules/sharepoint_file_transfer_folders_above_threshold.yaml:22]: Hi juju4, I will get back to you by 27 Oct 2023 on this.

Comment (juju4) [reply Solutions/Microsoft 365/Analytic Rules/sharepoint_file_transfer_folders_above_threshold.yaml:22]: Done

**Comment by juju4** [comment]
Tests all green now

**Review by v-rbajaj — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Comment by juju4** [comment]
Thanks!

**Pull request merged**